src/systems/filecoin_blockchain/struct/chain/chain.go

```diff
@@ -18,8 +18,17 @@ func (chain *Chain_I) TipsetAtEpoch(epoch abi.ChainEpoch) Tipset {
 
 // Draws randomness from the tipset at or immediately prior to `epoch`.
 func (chain *Chain_I) RandomnessAtEpoch(epoch abi.ChainEpoch) abi.RandomnessSeed {
-	ts := chain.TipsetAtEpoch(epoch)
-	return ts.MinTicket().DrawRandomness(epoch)
+	if epoch < genesis {
+		genesisTS := chain.TipsetAtEpoch(genesis)
+		genesisTix := genesisTS.MinTicket().DrawRandomness(genesis)
```

Comment:
What's this Ticket.DrawRandomness? It looks like that also mixes in an epoch number, but in this call, the fixed epoch 0.

IMO it would be clearer to consider the ticket as strictly a byte array, and do any epoch mixing in this code, not the ticket code.

Reply:
yes. fixed accordingly. Thank you!